python/cugraph-dgl/cugraph_dgl/cugraph_storage.py

```diff
@@ -0,0 +1,857 @@
+# Copyright (c) 2019-2022, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: Requires cuGraph nightly cugraph-22.12.00a220417 or later
+from cugraph_dgl.utils.cugraph_storage_utils import _assert_valid_canonical_etype
+from cugraph_dgl.utils.cugraph_storage_utils import convert_can_etype_s_to_tup
+from cugraph_dgl.utils.cugraph_storage_utils import backend_dtype_to_np_dtype_dict
+
+import dgl
+import dgl.backend as F
+from typing import Tuple
+
+
+class CuGraphStorage:
+    """
+    Duck-typed version of the DGLHeteroGraph class made for cuGraph
+    for storing graph structure and node/edge feature data.
+
+    This object is wrapper around cugraph's PropertyGraph and returns samples
+    that conform with `DGLHeteroGraph`
+    See: TODO link after https://github.com/rapidsai/cugraph/pull/2826
+
+
+    Read the user guide chapter :ref:`guide-`CuGraphStorage` for an in-depth
+    explanation about its usage.
+
+    """
+
+    def __init__(
+        self,
+        num_nodes_dict: dict,
+        single_gpu: bool = True,
+        cugraph_service_client=None,
+        device_id=0,
```

Comment:
Throughout this file too are cases of some args with type hints, some without. Should we be enforcing in reviews that these be present on new code, or would this be a FIXME type of situation for now?

Reply:
I have added all the type-hints that are possible without adding extra imports.  The missing ones are now `dgl objects` and `dataframe` objects that depends on the underlying library (for dgl bakcend, it can be pytorch, tensorflow,keras)  and for `dataframe` can be (`cuDF,dask_cuDF, pandas`).

Reply:
Have added issue: https://github.com/rapidsai/cugraph/issues/2922 to track it.